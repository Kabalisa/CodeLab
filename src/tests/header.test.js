import Header from '../components/header';

describe('header component test', () => {
  const props = { navigation: { navigate: jest.fn() } };
  it('renders without error', () => {
    const wrapper = shallow(<Header {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('call signout method', () => {
    const wrapper = shallow(<Header {...props} />);
    const button = wrapper.find('TouchableOpacity');
    button.props().onPress();
  });
  it('test onPress signout', () => {
    const wrapper = shallow(<Header {...props} />);
    wrapper.instance().signOut = jest.fn();
    wrapper.instance().forceUpdate();
    const button = wrapper.find('TouchableOpacity');
    button.props().onPress();
    expect(wrapper.instance().signOut).toHaveBeenCalled();
  });
});
