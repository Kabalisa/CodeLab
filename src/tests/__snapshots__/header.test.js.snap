// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header component test renders without error 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#5075D4",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "height": 85,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "flex-end",
        "bottom": 8,
        "color": "#5075D4",
        "fontSize": 24,
        "fontWeight": "bold",
        "left": 8,
      }
    }
  >
    CodeLab
  </Text>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "marginLeft": 240,
        "marginTop": 20,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#5075D4",
          "borderColor": "#5075D4",
          "borderWidth": 1,
          "padding": 4,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 16,
          }
        }
      >
        Log Out
      </Text>
    </View>
  </TouchableOpacity>
</View>
`;
